internal/resources/google/compute_cost_component_helpers.go

```diff
@@ -33,19 +33,21 @@ type ContainerNodeConfig struct {
 // computeCostComponent returns a cost component for Compute instance usage.
 func computeCostComponents(region, machineType string, purchaseOption string, instanceCount int64, monthlyHours *float64) ([]*schema.CostComponent, error) {
 	if strings.HasPrefix(strings.ToLower(machineType), "e2-custom") {
-		return nil, errors.New("Infracost currently does not support E2 custom instances")
+		return nil, errors.New("infracost currently does not support E2 custom instances")
 	}
 
 	sustainedUseDiscount := 0.0
 	fixPurchaseOption := ""
 
-	if strings.ToLower(purchaseOption) == "on_demand" {
+	if strings.ToLower(purchaseOption) == "on_demand" && monthlyHours != nil {
 		fixPurchaseOption = "OnDemand"
 		switch strings.ToLower(strings.Split(machineType, "-")[0]) {
 		case "c2", "n2", "n2d":
-			sustainedUseDiscount = 0.2
+			sustainedUseDiscount = getSustainedUseDiscount(monthlyHours, []float64{0, 0.25, 0.50, 0.75}, []float64{1.0, 0.8678, 0.733, 0.6}, []float64{0.2088, 0.1811, 0.1530, 0.1252})
```

Comment:
Can we add these values as constants and document where they were received from?

Reply:
I was considering storing them in a struct. Let me made this change and also add a comment about the values.